### DISCUSSION TITLE
Having trouble building the project

### DISCUSSION THREAD

**Discussion by vrrusso** [courselab/snaskii22#21] [Q&A] [answered]
Hi, I am currently trying to build the project, but when I try the first command on the tutorial:

```
./bootstrap.sh
```

I get:

```
-bash:  ./boostrap.sh: /bin/sh^M: bad interpreter: No such file or directory
```

What could be possibly wrong?

thank you all

**Comment by gp2112** [comment, marked as answer]
It looks like your system is not finding shell interpreter.
So:
- Which distribution and version (if not bleeding edge) are you using?
- Could you show us the output from `whereis sh`?

**Reply by vrrusso** [reply]
Ok!

- I am using Ubuntu 20.04.2 LTS on WSL 
- The output is as following:

```
sh: /usr/bin/sh /usr/share/man/man1/sh.1.gz
```

Thanks!

**Reply by gp2112** [reply]
So, the problem is `bootstrap.sh` uses shell interpreter that should be in /bin directory, as it is a System crucial software. I think this is something from WSL.

I don't know how WSL works, but the first thing I'd try is making a symbolic link to /usr/bin/sh on /bin.
```bash
ln -S /usr/bin/sh /bin/sh
```
Also, you could try running `bootstrap.sh` commands manually, but i'm not sure if it would work..
If none of this work, I'd recommend you running a VM with some Linux distro instead

**Reply by vrrusso** [reply]
Hey, it turns out I was building a previou version that was broken ... shame on me :)
